mlir/include/mlir/Dialect/Linalg/Transforms/Transforms.h

```diff
@@ -1395,6 +1395,24 @@ struct LinalgGeneralizationPattern
   }
 };
 
+struct LinalgSpecializationPattern
+    : public OpInterfaceRewritePattern<LinalgOp> {
+  using OpInterfaceRewritePattern<LinalgOp>::OpInterfaceRewritePattern;
```

Comment:
This can just be an `OpRewritePattern<GenericOp>` instead of doing a cast from LinalgOp.

Reply:
Awesome suggestion. simplifies. Fixed.